src/firmware/boards/powerboard_esp32/PowerLoop.cpp

```diff
@@ -0,0 +1,275 @@
+int G_MOTOR_SPEED_ADDRESS;
+int BREAKBEAM_INTER_ADDRESS;
+int BREAKBEAM_DIFF_ADDRESS;
+int CHICKER_AUTOKICK_ADDRESS;
+int CHICKER_AUTOCHIP_ADDRESS;
+int KICK_DISTANCE_SPEED_ADDRESS;
+int CHIP_DISTANCE_SPEED_ADDRESS;
+int PULSE_WIDTH_ADDRESS;
+int FLYBACK_FAULT_ADDRESS;
+
+//Others
+int COUNTER;
+float checker;
+
+int hv_data;
+int g_encoder_angle;
+float g_motor_angle;
+float g_motor_speed;
+int breakbeam_diff;
+int breakbeam_inter;
+int autokick;
+int autochip;
+float kick_distance_speed;
+float chip_distance_speed;
+int flyback_fault;
+
+
+
+void setup(){
+
+    //Initializing serial ports 
+    Serial.begin(9600);
+    SUART.begin(9600);
+
+    //Initializing GPIO pins
+    pinMode(HV_MEASUREMENT_PIN, INPUT);
+    pinMode(G_ENCODER_PIN, INPUT);
+    pinMode(G_MOTOR_ANGLE_PIN, INPUT);
+    pinMode(G_MOTOR_SPEED_PIN, INPUT);
+    pinMode(BREAKBEAM_DIFF_PIN, INPUT);
+    pinMode(BREAKBEAM_INTER_PIN, INPUT);
+    pinMode(CHICKER_AUTOCHIP_PIN, INPUT);
+    pinMode(CHICKER_AUTOKICK_PIN, INPUT);
+    pinMode(KICK_DISTANCE_SPEED_PIN, INPUT);
+    pinMode(CHIP_DISTANCE_SPEED_PIN, INPUT);
+    pinMode(PULSE_WIDTH_PIN, INPUT);
+    pinMode(FLYBACK_FAULT_PIN, INPUT);
+
+    
+    // initialize EEPROM with predefined size
+    for (int i = 0 ; i < EEPROM.length() ; i++) {
+      EEPROM.write(i, 0);
+    }
+
+}
+
+
+void loop(){
+    INITIAL_MESSAGE = SUART.available(); //Checks if initial message is avaliable (If a message is recieved)
+
+    if (INITIAL_MESSAGE != 0) //If message avaliable, continue
+    {
+        
+        MESSAGE = SUART.read(); //Store entire message
+
+        if (MESSAGE == HV_MEASUREMENT){
+
+            //Action
+            for(COUNTER = 0; COUNTER < 5; COUNTER++){  //Total 2.5s of reading in 0.5 intervals
+
+                hv_data = analogRead(HV_MEASUREMENT_PIN);
+
+                Serial.println(hv_data);
+
+                delay(500); //wait 0.5s
+             
+            }
+
+            EEPROM.write(HV_MEASUREMENT_ADDRESS, hv_data);
+
+        }
+
+        if(MESSAGE == GENEVA_ENCODER){
+
+            //Action
+            g_encoder_angle = analogRead(G_ENCODER_PIN);
+
+            Serial.println(g_encoder_angle);
+
+            EEPROM.write(G_ENCODER_ADDRESS, g_encoder_angle);
+        }
+
+        if(MESSAGE == GENEVA_MOTOR_ANGLE){
+
+            //Action
+            byte checker = SUART.available(); //check if a character has arrived via SUART Port
+            if (checker != 0) //a charctaer has arrived; it has been auto saved in FIFO; say 1 as 0x31
+            {
+                g_motor_angle = SUART.read(); //read arrived character from FIFO (say 1) and put into x as 0x31 
+
+                analogWrite(G_MOTOR_ANGLE_PIN, g_motor_angle); //Set motor angle
+            }
+            else{
+                Serial.println("Failed Retrieve Geneva Motor Angle");
+            }
+
+
+            EEPROM.write(G_MOTOR_ANGLE_ADDRESS, g_motor_angle);
+        }
+
+        if(MESSAGE == GENEVA_MOTOR_SPEED){

```

Comment:
Can we read geneva motor speed? I guess it would be useful to know if the geneva motor is moving (i.e its repositioning itself) but we don't need the exact RPM (but im not sure we can even read that)

Reply:
I don't believe we can read the geneva motor speed as we do not have an output designed so on the powerboard.

The condition you in line 155 is to assign the geneva motor a specific speed